src/profilePage/RepoCard.tsx

```diff
@@ -1,10 +1,65 @@
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import { octokit } from "../../environment/apiKey";
+import { hexColors } from "../Models/data";
+import {
+  BranchInfo,
+  CommitItem,
+  LanguageData,
+  RepoItem,
+} from "../Models/interfaces";
+import { getCommits, getLanguages, getBranches } from "./Api/profileApi";
+
 interface RepoCardProps {
   repoName: string;
+  repoInfo: RepoItem;
+  sendUpLangauges: (languages: LanguageData) => void;
+  sendUpCommits: (commits: CommitItem[]) => void;
 }
 
-const RepoCard: React.FC<RepoCardProps> = ({ repoName }) => {
+const RepoCard: React.FC<RepoCardProps> = ({
+  repoName,
+  repoInfo,
+  sendUpLangauges,
+  sendUpCommits,
+}) => {
+  const [branches, setBranches] = useState<BranchInfo[]>([]);
+  const [branchNumber, setBranchNumber] = useState<number>();
+  const [languages, setLanguages] = useState<LanguageData>({});
+  const [commits, setCommits] = useState<LanguageData[]>([]);
+
+  useEffect(() => {
+    getBranches(repoInfo, setBranches, setBranchNumber);
+    getLanguages(repoInfo, setLanguages, sendUpLangauges);
+    getCommits(repoInfo, setCommits, sendUpCommits);
+  }, []);
```

Comment:
The confusion I have where you have tanstack query setup for this, but disable it...